scripts/tidy.sh

```diff
@@ -2,7 +2,25 @@
 
 set -euo pipefail
 
-for f in $(git ls-files | grep go.mod)
+ROOT_DIR=$(git rev-parse --show-toplevel)
+SDK_DIR="$ROOT_DIR/sdk"
+PKG_DIR="$ROOT_DIR/pkg"
```

Comment:
```suggestion
readonly ROOT_DIR=$(git rev-parse --show-toplevel)
readonly SDK_DIR="$ROOT_DIR/sdk"
readonly PKG_DIR="$ROOT_DIR/pkg"
```

Reply:
What does the readonly keyword do, and did I miss a shellcheck check?

Reply:
It makes the variable read-only.
shellcheck doesn't check for it, but if you write to a readonly var after, you'll get an error:

```
readonly FOO=bar
FOO=baz
# error: read-only variable: FOO
```

Reply:
Got it, wondered if it did anything other than what it said on the tin. I've not heard of this as a best practice before.

Reply:
In all of our shell scripts, I could only find one readonly var declared and it looks like shellcheck is unhappy with it. Adding `readonly` here makes the script longer and less straightforward by triggering SC2155:

![Screenshot of `readonly ROOT_DIR` triggering a shellcheck lint for SC2155: declare and assign separately.](https://user-images.githubusercontent.com/788800/214374358-deda8012-5c72-4b8a-b3b5-bd06c34e1b83.png)

And if that is followed, you :

```sh
readonly ROOT_DIR
ROOT_DIR="$(git rev-parse --show-toplevel)"
```


Which errors when run, so it appears the fix is:

```sh
ROOT_DIR=$(git rev-parse --show-toplevel)
declare -r ROOT_DIR
```

IMO: this change has negative utility in making the bash script readable

Reply:
Gotcha, it's fine by me to skip adding `readonly` since there's negative utility.

In the future, I'd wrap code like this in a bash function, then use `local -r` where possible. That gives the variable local scope in addition to immutability :D Not sure how shellcheck feels about that, but there's a section on `local` misuse in SC2155 (the same lint you referenced).

Reply:
I lied, there's a note on this at the top of the page:

> Note that ShellCheck does not warn about masking of local read-only variables, such as local -r foo=$(cmd), even though this also masks the return value. This is because the alternative local foo; foo=$(cmd); local -r foo is repetitive and cumbersome. To see warnings for this and many other additional cases of suppressed exit codes, enable check-extra-masked-returns.